### PR TITLE
feat: add type annotations in jina executors children

### PR DESCRIPTION
**Pull request by rameshwara**
feat: add type annotations in jina executors children

### CODE REVIEW

**Comment by JoanFM** [comment]
Hey @rameshwara can you please divide this PR further? It lets us have proper review of the code

**Comment by rameshwara** [comment]
> Hey @rameshwara can you please divide this PR further? It lets us have proper review of the code

There are not many additions made actually and hence, I put them all under a single PR. But if it is still not enough, then I guess, this pull request needs to be closed and I would submit a separate PR for each of the executor sub-modules.

**Comment by rameshwara** [comment]
Hi @JoanFM, let me know once you finish reviewing the annotations in jina executors encoders. I shall then commit the annotations in jina executors indexers and wait for your review on that, before I commit annotations in jina executors rankers. And can you close and merge this PR once you finish reviewing annotations in rankers?

**Review by JoanFM — APPROVED** [review]

**Pull request merged**
